server/src/main/java/org/eclipse/openvsx/ExtensionProcessor.java

```diff
@@ -227,11 +232,21 @@ public ExtensionLicense getLicense(ExtensionVersion extension) {
         if (result == null) {
             return null;
         }
+        licenseId = this.getLicenseId(result.getFirst());
+        extension.setLicense(licenseId);
```

Comment:
In this case the detection should run only if `extension.getLicense()` is null or empty.